### ISSUE TITLE
Auto-indent fails to break out of nested context

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#3686]
Manual transfer from https://github.com/microsoft/pyrx/issues/2956

```python
dataSetSize = 10

input = []
output = []
for i in range(10):
    number = random.random()
    input.append(number)
    if number < 0.5:
        output.append(0)
    else:
        output.append(1)
|<- hit enter here
		|<-jumps here
|<- expected here
```

**Comment by debonte** [comment]
Simpler repro:

```python
if True:
    x = 1
|<- hit enter here
```

It doesn't repro for me if the `|<- hit enter here` line is the last line in the file.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221141-30-November-2022-PreRelease)
